### PR TITLE
Use primary constructors

### PR DESCRIPTION
**Pull request by martincostello**
- Use primary constructors from C# 12 in a few places where relevant.
- Apply some IDE suggestions to use newer C# syntax from previous new releases of C#.
- Fix typo.
- Update .NET SDK for RC1.

If the new syntax isn't controversial, I'll look at applying it to the OAuth providers too.

### CODE REVIEW

File: src/AspNet.Security.OpenId/OpenIdAuthenticationInitializer.cs [hunk 1]
```diff
@@ -17,21 +17,14 @@ namespace Microsoft.Extensions.DependencyInjection;
 /// Contains the methods required to ensure that the configuration used by
 /// the OpenID 2.0 generic handler is in a consistent and valid state.
 /// </summary>
+/// <remarks>
```
Comment (kevinchalet) [inline src/AspNet.Security.OpenId/OpenIdAuthenticationInitializer.cs:20]: Hum, that looks weird. Is it suggested by a Roslyn code fixer?

File: src/AspNet.Security.OpenId/OpenIdAuthenticationInitializer.cs [hunk 2]
```diff
@@ -17,21 +17,14 @@ namespace Microsoft.Extensions.DependencyInjection;
...
     where THandler : OpenIdAuthenticationHandler<TOptions>
 {
-    private readonly IDataProtectionProvider _dataProtectionProvider;
-
-    /// <summary>
-    /// Creates a new instance of the <see cref="OpenIdAuthenticationInitializer{TOptions, THandler}"/> class.
-    /// </summary>
-    public OpenIdAuthenticationInitializer([NotNull] IDataProtectionProvider dataProtectionProvider)
-    {
-        _dataProtectionProvider = dataProtectionProvider;
```
Comment (kevinchalet) [inline src/AspNet.Security.OpenId/OpenIdAuthenticationInitializer.cs:32]: FWIW, I personally tend to null-check all the arguments used in the "DI constructors", so I'd probably keep the field but with a null check (`private readonly IDataProtectionProvider _dataProtectionProvider = dataProtectionProvider ?? throw new ArgumentNullException(dataProtectionProvider);`).

That said, I don't see arguments of DI constructors guarded anywhere in the aspnet-contrib projects, so... your call 😄

**Review by kevinchalet — APPROVED** [review]
👍🏻

File: src/AspNet.Security.OpenId/OpenIdAuthenticationInitializer.cs (hunk 1 above)
Comment (martincostello) [reply src/AspNet.Security.OpenId/OpenIdAuthenticationInitializer.cs:20]: Yeah the code-fixer did that. I _assumed_ it was correct and is the way you document a constructor if using primary constructors.

File: src/AspNet.Security.OpenId/OpenIdAuthenticationInitializer.cs (hunk 2 above)
Comment (martincostello) [reply src/AspNet.Security.OpenId/OpenIdAuthenticationInitializer.cs:32]: I'll leave it as it is as it hasn't changed the existing behaviour and no-one's complained about it before.

File: src/AspNet.Security.OpenId/OpenIdAuthenticationInitializer.cs (hunk 1 above)
Comment (martincostello) [reply src/AspNet.Security.OpenId/OpenIdAuthenticationInitializer.cs:20]: If this is wrong, let me know and I'll update post-merge.

Comment (kevinchalet) [reply src/AspNet.Security.OpenId/OpenIdAuthenticationInitializer.cs:20]: Looks too weird to my taste, let's remove these <remarks> nodes, they don't have anything super useful anyway 😄

**Pull request merged**